Root relative path is not resolved using the resolve.roots

According to the documentation at https://github.com/webpack-contrib/html-loader#roots server-relative URLs (starting with '/') should be resolved if resolve.roots is specified. But this is not working.

- Operating System: Windows 10
- Node Version: 12.18.0
- NPM Version: 6.14.4
- webpack Version: 5.30.0
- html-loader Version: 2.1.2

### Expected Behavior

Resolve server-relative URL by using the root specified by resolve.roots

### Actual Behavior

Error is raised: Cannot find module '/img/image1.jpg' from [...]

### Code

### Webpack Config
<pre>
resolve: {
          roots: [
                path.resolve(__dirname, 'src')
          ]
}
</pre>

<pre>
{
                  loader: "html-loader",
                  options: {
                      esModule: false,
                      minimize: false,
                      sources: {
                          list: [
                              {tag: 'img', attribute: 'src', type: 'src'},
                          ]
                      }
                 }
}
</pre>

### HTML File:
<pre>
    &lt;img src="/img/image1.jpg"/&gt;
</pre>

### Directory Structure:
<pre>
project
    src
       img
          image1.jpg
    webpack.config
</pre>

--- a/test/fixtures/resolve-roots.html
+++ b/test/fixtures/resolve-roots.html
@@ -1,0 +1,1 @@
+<img src="/image3.png"/>

--- a/test/fixtures/resolve-roots.js
+++ b/test/fixtures/resolve-roots.js
@@ -1,0 +1,3 @@
+import html from './resolve-roots.html';
+
+export default html;

--- a/test/sources-option.test.js
+++ b/test/sources-option.test.js
@@ -85,7 +85,7 @@ describe("'sources' option", () => {
     expect(getErrors(stats)).toMatchSnapshot("errors");
   });
 
-  it("should allow to add more attributes ti default values", async () => {
+  it("should allow to add more attributes to default values", async () => {
     const compiler = getCompiler("simple.js", {
       sources: {
         list: [
@@ -324,6 +324,28 @@ describe("'sources' option", () => {
     const stats = await compile(compiler);
 
     expect(getModuleSource("./simple.html", stats)).toMatchSnapshot("module");
+    expect(
+      execute(readAsset("main.bundle.js", compiler, stats))
+    ).toMatchSnapshot("result");
+    expect(getWarnings(stats)).toMatchSnapshot("warnings");
+    expect(getErrors(stats)).toMatchSnapshot("errors");
+  });
+
+  it("should work and supports `resolve.roots`", async () => {
+    const compiler = getCompiler(
+      "resolve-roots.js",
+      {},
+      {
+        resolve: {
+          roots: [path.resolve(__dirname, "fixtures/nested/")],
+        },
+      }
+    );
+    const stats = await compile(compiler);
+
+    expect(getModuleSource("./resolve-roots.html", stats)).toMatchSnapshot(
+      "module"
+    );
     expect(
       execute(readAsset("main.bundle.js", compiler, stats))
     ).toMatchSnapshot("result");

